setup-r-dependencies/action.yaml

```diff
@@ -120,6 +124,19 @@ runs:
         shell: Rscript {0}
         working-directory: ${{ inputs.working-directory }}
 
+      # Step used to help speed up package installation on troublesome Actions
+      # If an Action fails, cache from `actions/cache@v3` will NOT be saved
+      # However `actions/cache/save@v3` will always save, warming the next execution of `actions/cache@v3` above
+      - name: Save R package cache
+        # If `actions/cache@v3` restored from `key`, then `cache-hit == 'true'`; Skip this step
+        if: inputs.cache == 'true' && steps.r-pkg-cache.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v3
+        with:
+          path: |
+            ${{ env.R_LIBS_USER }}/*
+            !${{ env.R_LIBS_USER }}/pak
+          key: ${{ format('{0}-{1}-{2}-init-{3}', steps.install.outputs.os-version, steps.install.outputs.r-version, inputs.cache-version, hashFiles(format('{0}/.github/pkg.lock', inputs.working-directory ))) }}
```

Comment:
Why not save to the original key? Why do we want to use another set of caches?

Reply:
I will give it a try! I remember seeing something about the key already existing, but with the `if` statement just above, we should be safe.

Reply:
It might fail if a concurrent workflow creates this key before us, so we definitely should not fail if the key happens to exist.

Reply:
It works, but it will produce a message on the initial successful run saying that the cache already exists.

### Example workflow

Workflow on cold cache:
* actions/cache@v3 - cache miss
* actions/cache/save@v3 - saves cache
* Successful Action
* Post step: actions/cache@v3 - produces message about key already existing

Workflow on warmed cache:
* actions/cache@v3 - cache hit
* actions/cache/save@v3 - (skipped)
* Successful Action
* Post step: actions/cache@v3 - success


### Example usage

* Initial Run
	* Success
		* Cache saved to original key location on initial successful run: https://github.com/rstudio/htmltools/actions/runs/3971016288/jobs/6807579837#step:3:4338
		* Message produced on initial successful run: https://github.com/rstudio/htmltools/actions/runs/3971016288/jobs/6807579837#step:14:6
	* Failure
		* Cache saved to original key location on initial failing run: https://github.com/rstudio/htmltools/actions/runs/3971016288/jobs/6807579108#step:3:2943
* Followup Run
	* Success
		* Cache restored: https://github.com/rstudio/htmltools/actions/runs/3971016288/jobs/6807825306#step:3:3734
		* Save step skipped: https://github.com/rstudio/htmltools/actions/runs/3971016288/jobs/6807825306#step:3:4018
		* Cache not saved as it was a cache hit: https://github.com/rstudio/htmltools/actions/runs/3971016288/jobs/6807825306#step:14:5
	* Failure
		* Cache restored: https://github.com/rstudio/htmltools/actions/runs/3971016288/jobs/6807824705#step:3:2694

Reply:
Going to implement the PR with actions/cache/restore@v3. 

Resolving convo as the second cache is removed and no confusing messages are produced.